File: frontend/lib/models/user_course.dart
class UserCourse {
  final String userId;
  final int courseId;
  final String term;
  final int year;
  final String? startTime;
  final String? endTime;
  final String? createdAt;

  UserCourse({
    required this.userId,
    required this.courseId,
    required this.term,
    required this.year,
    this.startTime,
    this.endTime,
    this.createdAt,
  });

  factory UserCourse.fromJson(Map<String, dynamic> json) {
    return UserCourse(
      userId: json['user_id'] as String,
      courseId: json['course_id'] as int,
      term: json['term'] as String,
      year: json['year'] as int,
      startTime: json['start_time'] as String?,
      endTime: json['end_time'] as String?,
      createdAt: json['created_at'] as String?,
    );
  }
}

File: backend/app/schemas/user_course.py
from pydantic import BaseModel, Field
from typing import Optional
from datetime import datetime


class UserCourseCreate(BaseModel):
    course_id: int
    term: str = Field(..., description="e.g., 'Fall', 'Spring', 'Summer'")
    year: int = Field(..., description="e.g., 2026")
    start_time: Optional[str] = Field(None, description="e.g., '09:00'")
    end_time: Optional[str] = Field(None, description="e.g., '10:30'")


class UserCourseResponse(BaseModel):
    user_id: str
    course_id: int
    term: str
    year: int
    start_time: Optional[str] = None
    end_time: Optional[str] = None
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None

    class Config:
        from_attributes = True


class UserCourseUpdate(BaseModel):
    term: Optional[str] = None
    year: Optional[int] = None


User_Course_Create = UserCourseCreate
User_Course_Response = UserCourseResponse
